### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by nquesada**
This PR does 3 things:

1. It updates the imports from the latest master branch of The Walrus after the function in `quantum` have been renamed.
2. Adds thresholding for 2 dimensional probability mass functions.
3. Adds the noise reduction factor as one of the outputs of `marginal_calcs_2d`.

This PR should only be merged after a new release of The Walrus is made.

### CODE REVIEW

**Comment by nquesada** [comment]
This PR closes https://github.com/XanaduAI/sqtom/issues/8

File: sqtom/fitting_2d.py [hunk 1]
```diff
@@ -85,15 +85,19 @@ def marginal_calcs_2d(jpd_data, as_dict=True):
     g2s = (ns2 - ns) / ns ** 2
     g2i = (ni2 - ni) / ni ** 2
     g11 = (na @ jpd_data @ nb) / (ns * ni)
+    nrf = np.sum([[((i - j) ** 2) * jpd_data[i, j] for i in range(inta)] for j in range(intb)])
```
Comment (antalszava) [inline sqtom/fitting_2d.py:88]: Worth adding to the docstring that this stands for the noise reduction factor (`Returns`)

File: sqtom/fitting_2d.py [hunk 2]
```diff
@@ -114,15 +118,43 @@ def gen_hist_2d(beam1, beam2):
     return mass_fun
 
 
+def threshold_2d(ps, nmax, mmax):
+    """Thresholds a 2D probability distribution by assigning events with more than nmax (mmax) photons
+    in the first (second) axis to the nmax (mmax) bin.
+
+    Args:
+        ps (array): probability distribution
+        nmax (int): threshold value
```
Comment (antalszava) [inline sqtom/fitting_2d.py:127]: `mmax` is missing here

File: sqtom/fitting_2d.py [hunk 3]
```diff
@@ -114,15 +118,43 @@ def gen_hist_2d(beam1, beam2):
...
+    sq_label="sq_",
+    noise_label="noise",
 ):
     """Returns a model fit from the parameter guess and the data
 
     Args:
         pd_data (array): one dimensional array of the probability distribution of the data
         guess (dict): dictionary with the guesses for the different parameters
         method (string): method to be used by the optimizer
+        thresholf (boolean or list): threshold photon number for the signal and idlers
```
Comment (antalszava) [inline sqtom/fitting_2d.py:157]:
```suggestion
        threshold (boolean or list): threshold photon number for the signal and idlers
```

File: sqtom/fitting_2d.py [hunk 4]
```diff
@@ -114,15 +118,43 @@ def gen_hist_2d(beam1, beam2):
...
+        probs[i, mmax - 1] += np.sum(ps[i, mmax:])
+    probs[nmax - 1, mmax - 1] = np.sum(ps[nmax - 1 :, mmax - 1 :])
+    return probs
+
+
 def fit_2d(
-    pd_data, guess, do_not_vary=[], method="leastsq", cutoff=50, sq_label="sq_", noise_label="noise"
+    pd_data,
+    guess,
+    do_not_vary=[],
```
Comment (antalszava) [inline sqtom/fitting_2d.py:144]: (Minor suggestion): might be worth adjusting this as done for [`fit_1d`](https://github.com/XanaduAI/sqtom/blob/748bb2cee9c0e39c85c0e934eecc513eec413188/sqtom/fitting_1d.py#L75) (set to `None` and then creating an empty list later)

File: sqtom/tests/test_fitting_1d.py [hunk 5]
```diff
@@ -31,7 +31,7 @@ def test_exact_model_1d(n_modes, threshold, do_not_vary):
     params["n_modes"] = n_modes
     params["eta"] = eta
     params["noise"] = noise
-    if threshold is not False:
+    if threshold:
```
Comment (antalszava) [inline sqtom/tests/test_fitting_1d.py:34]: :100:

File: sqtom/tests/test_fitting_2d.py [hunk 6]
```diff
@@ -94,5 +98,6 @@ def test_marginal_calcs_2d():
     nmean = 1.0
     ps = twinbeam_pmf({"n_modes": 1.0, "sq_0": nmean})
     assert np.allclose(
-        marginal_calcs_2d(ps, as_dict=False), np.array([nmean, nmean, 2 + 1 / nmean, 2, 2]),
+        marginal_calcs_2d(ps, as_dict=False),
+        np.array([nmean, nmean, 2 + 1 / nmean, 2, 2, 0]),
```
Comment (antalszava) [inline sqtom/tests/test_fitting_2d.py:102]: (Minor suggestion): for readability, the result and the expected object could be defined prior to calling `np.allclose`, for example:
```python
res = marginal_calcs_2d(ps, as_dict=False)
expected = np.array([nmean, nmean, 2 + 1 / nmean, 2, 2, 0])
assert np.allclose(res, expected)
```

**Review by antalszava — APPROVED** [review]
Looks good to me! :100: Mostly checked code quality, had a couple of minor suggestions and questions, but nothing major

File: sqtom/fitting_1d.py [hunk 7]
```diff
@@ -120,7 +120,7 @@ def fit_1d(
         def model_1d(params, pd_data):
             ndim = pd_data.shape[0]
             dpmf = degenerate_pmf(params, cutoff=cutoff)
-            return threshold_1d(dpmf, ndim) - pd_data
+            return threshold_1d(dpmf, ndim) - threshold_1d(pd_data, ndim)
```
[the remark this answers is not in the document]
Comment (nquesada) [reply sqtom/fitting_1d.py:123]: Good point. It is not needed.

File: sqtom/fitting_2d.py (hunk 1 above)
Comment (nquesada) [reply sqtom/fitting_2d.py:88]: Yup, good point.

File: sqtom/fitting_2d.py (hunk 2 above)
Comment (nquesada) [reply sqtom/fitting_2d.py:127]: Thanks. Good catch!

**Pull request merged**